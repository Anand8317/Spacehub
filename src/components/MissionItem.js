import React from 'react';
import PropTypes from 'prop-types';
import Button from 'react-bootstrap/Button';
import { useDispatch } from 'react-redux';
import { joinMissionAction, leaveMissionAction } from '../redux/missions/missions';

const MissionItem = (props) => {
  const {
    id, missionName, missionDesc, reserved,
  } = props;
  const dispatch = useDispatch();

  const clickHandler = (e) => {
    if (!reserved) {
      dispatch(joinMissionAction(e.target.parentNode.parentNode.id));
      return;
    }

    dispatch(leaveMissionAction(e.target.parentNode.parentNode.id));
  };

  return (
    <tr id={id}>
      <td className="missionName">{missionName}</td>
      <td className="missionDesc">{missionDesc}</td>
      <td width={150} className="btn-container btn-container">
        { reserved ? <Button variant="primary" className="am-btn center btn" size="sm">ACTIVE MEMBER</Button> : <Button variant="secondary" className="center btn" size="sm">NOT A MEMBER </Button> }
      </td>
      <td width={150} className="btn-container btn-container">
        { reserved ? <Button variant="outline-danger" className="center btn" size="sm" onClick={clickHandler}>LEAVE MISSION</Button> : <Button variant="outline-secondary" className="center btn" size="sm" onClick={clickHandler}>JOIN MISSION</Button> }
      </td>
    </tr>
  );
};

MissionItem.propTypes = {
  id: PropTypes.string.isRequired,
  missionName: PropTypes.string.isRequired,
  missionDesc: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default MissionItem;
