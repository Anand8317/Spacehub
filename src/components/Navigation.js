import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/logo.png';

const Navigation = () => (
  <div>

    <nav className="navbar">
      <div className="links">
        <div className="logo">
          <img src={logo} className="logo-pic" alt="spacehub-logo" />
          <h1 className="logo-title">Space Travelers&apos Hub</h1>
        </div>
        <ul>
          <li><NavLink className={({ isActive }) => (isActive ? 'link-active' : 'link')} to="/">Rockets</NavLink></li>
          <li><NavLink className={({ isActive }) => (isActive ? 'link-active' : 'link')} to="/missions">Missions</NavLink></li>
          <li><NavLink className={({ isActive }) => (isActive ? 'link-active' : 'link')} to="/profile">My profile</NavLink></li>
        </ul>
      </div>
    </nav>

  </div>
);

export default Navigation;
