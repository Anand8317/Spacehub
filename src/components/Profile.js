import React from "react";
import Table from "react-bootstrap/Table";
import { useSelector } from "react-redux";

const MissionProfile = () => {
  const mission = useSelector((state) => state.missions);

  return (
    <div className="mission-table">
      <h2>My Missions</h2>
      <Table bordered>
        <tbody>
          {mission
            .filter((obj) => obj.reserved)
            .map((item) => (
              <tr key={item.mission_id}>
                <td>{item.mission_name}</td>
              </tr>
            ))}
        </tbody>
      </Table>
    </div>
  );
};

const RocketsProfile = () => {
  const rockets = useSelector((state) => state.rockets.rocket);
  const reservedRocket = Array.isArray(rockets) ? rockets.filter((rocket) => rocket.reserved) : [];
  console.log(reservedRocket);
  return (
    <div className="rocket-table">
      <h2>My Rockets</h2>
      <Table bordered>
        <tbody>
          {
            reservedRocket.map((rocket) => (
              <tr key={rocket.id}>
                <td>{rocket.name}</td>
              </tr>
            ))}
      
        </tbody>
      </Table>
    </div>
  );
};

export { RocketsProfile, MissionProfile };
