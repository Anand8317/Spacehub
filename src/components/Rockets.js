import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { reserveRocket } from '../redux/rockets/rockets';

const RocketsList = ({
  id, title, description, image, reserved, confirmReservation,
}) => {
  const dispatch = useDispatch();
  const HandlerBtn = () => {
    dispatch(reserveRocket(id));
  };
  return (
    <div>
      <div className="rocketContainer">
        <img src={image} alt={title} />
        <div className="rocketInfo">
          <h3>{title}</h3>
          {
            reserved
              ? (
                <p>
                  <span id="reserved">{confirmReservation}</span>
                  {description}
                </p>
              )
              : (
                <p>
                  <span id="not-reserved">{confirmReservation}</span>
                  {description}
                </p>
              )
          }
          {
            reserved
              ? <button className="reserveBtn" id="cancel" type="button" onClick={HandlerBtn}>Cancel Reservation</button>
              : <button className="reserveBtn" id="reserve" type="button" onClick={HandlerBtn}>Reserve Rocket</button>
          }
        </div>
      </div>
    </div>
  );
};

RocketsList.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  confirmReservation: PropTypes.string.isRequired,
};

export default RocketsList;
