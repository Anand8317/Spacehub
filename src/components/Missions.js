import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Table from 'react-bootstrap/Table';
import { fetchMissonsAction } from '../redux/missions/missions';
import MissionItem from './MissionItem';

let flag = false;
const Mission = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    if (!flag) {
      dispatch(fetchMissonsAction());
      flag = true;
    }
  }, [dispatch]);

  const mission = useSelector((state) => state.missions);

  return (
    <Table className="table1" striped bordered hover responsive="sm">
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th aria-label="member" />
        </tr>
      </thead>
      <tbody>
        {mission.map((item) => (
          <MissionItem
            key={item.mission_id}
            id={item.mission_id}
            missionName={item.mission_name}
            missionDesc={item.description}
            reserved={item.reserved ? item.reserved : false}
          />
        ))}
      </tbody>
    </Table>

  );
};

export default Mission;
