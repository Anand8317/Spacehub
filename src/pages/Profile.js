import React from 'react';
import { MissionProfile, RocketsProfile } from '../components/Profile';

const Profile = () => (
  <div className="profile-container">
    <MissionProfile />
    <RocketsProfile />
  </div>

);

export default Profile;
