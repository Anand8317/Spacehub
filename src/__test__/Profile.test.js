import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { RocketsProfile } from '../components/Profile';
import store from '../redux/configureStore';

it('rockets list renders properly', () => {
  const tree = renderer
    .create(<Provider store={store}><RocketsProfile /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
