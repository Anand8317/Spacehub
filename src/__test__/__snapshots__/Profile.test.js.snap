// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rockets list renders properly 1`] = `
<div
  className="rocket-table"
>
  <h2>
    My Rockets
  </h2>
  <table
    className="table table-bordered"
  >
    <tbody />
  </table>
</div>
`;
